admin-dev/themes/new-theme/js/pages/product/edit/combinations-manager.js

```diff
@@ -27,37 +27,101 @@ import ProductMap from '@pages/product/product-map';
 import CombinationsGridRenderer from '@pages/product/edit/combinations-grid-renderer';
 import CombinationsService from '@pages/product/services/combinations-service';
 import DynamicPaginator from '@components/pagination/dynamic-paginator';
+import SubmittableInput from '@components/form/submittable-input';
 
 const {$} = window;
 
 export default class CombinationsManager {
   constructor() {
     this.$productForm = $(ProductMap.productForm);
+    this.$combinationsContainer = $(ProductMap.combinations.combinationsContainer);
+    this.combinationIdInputsSelector = ProductMap.combinations.combinationIdInputsSelector;
     this.initialized = false;
+    this.combinationsService = new CombinationsService(this.getProductId());
+
     this.init();
 
     return {};
   }
 
   init() {
-    const productId = this.getProductId();
     this.paginator = new DynamicPaginator(
       ProductMap.combinations.paginationContainer,
-      new CombinationsService(productId),
+      this.combinationsService,
       new CombinationsGridRenderer(),
     );
-
+    this.initSubmittableInputs();
+    this.onDefaultCombinationSelection();
     // Paginate to first page when tab is shown
     this.$productForm.find(ProductMap.combinations.navigationTab).on('shown.bs.tab', () => this.firstInit());
   }
 
+  initSubmittableInputs() {
+    const combinationToken = this.getCombinationToken();
+
+    new SubmittableInput('.combination-quantity', async (input) => {
+      await this.combinationsService.updateListedCombination(
+        this.findCombinationId(input),
+        {'combination_item[quantity][value]': input.value, 'combination_item[_token]': combinationToken},
```

Comment:
Maybe these names should be placed in a dedicated `CombinationItemMap` ?

```suggestion
        {`${CombinationItemMap.quantityInput}`: input.value, `${CombinationItemMap.token}`: combinationToken},
```

Reply:
That might suggest we could have dedicated directory for combinations, that would make sense, but if not, then I would say we can put it in same ProductMap

Reply:
Mmm, indeed a folder for combination makes sense, since we're gonna have more and more components for them And since they are gonna bring even more input mapping, that don't interact with each other, splitting them into two different mapping makes sense

Reply:
ok so for now I've added them as usual to productMap. I don't want to extend this PR, we can introduce combinations dir in another one.